Sources/AppcuesKit/Presentation/ExperienceLoader.swift

```diff
@@ -42,6 +49,15 @@ internal class ExperienceLoader: ExperienceLoading {
                 self?.config.logger.error("Loading experience %{public}s failed with error %{public}s", experienceID, "\(error)")
                 completion?(.failure(error))
             }
+
+            self?.lastPreviewExperienceID = published ? nil : experienceID
```

Comment:
just wondering if this should also clear out some way when the preview is dismissed - i.e. only allow shake to refresh when preview is up?  thinking of this use case: preview something, dismiss it, do other stuff for 10 mins, trigger a shake gesture - preview comes back.  This is a pretty far edge case of concern though.

Reply:
The shake gesture is only detected in the `ExperienceStepViewController` so once that's out the of the responder chain, the shake detection is gone too. If different experience has loaded then this will have reset. So it's not the most direct mechanism, but it should be safe.